src/pymor/parallel/ipython.py

```diff
@@ -146,10 +148,15 @@ def __init__(self, num_engines=None, **kwargs):
         self.view.map_sync(_setup_worker, range(len(self.view)))
         self._remote_objects_created = Counter()
 
+        if defaults.defaults_changes() > 0:
+            self._update_defaults()
+
     def __len__(self):
         return len(self.view)
 
     def _push_object(self, obj):
+        if defaults.defaults_changes() > self._updated_defaults:
+            self._update_defaults()
```

Comment:
I think this should go in `_apply`, `_apply_only` and `_map`. `_push_object` does not execute any actual code, but before an computation is initiated via these methods, we have to check the defaults.

Reply:
Do you want to modify the defaults within the algorithms? Otherwise it should be enough to set them once in `_push_object` or `_map`? Adapted it anyway. Some additional checks should not impact the performance much here, but your suggested behavior could be more intuitive.

Reply:
This is not changing defaults within an algorithm. It's possible to use a `WorkerPool` without ever calling `push_object`. You only call `push_object` when you want to avoid to repeatedly communicate the same large object to the workers.